### DISCUSSION TITLE
dpl mode / settings via web-api

### DISCUSSION THREAD

**Discussion by mastertape** [hoylabs/OpenDTU-OnBattery#1284] [Q&A]
Hallo zusammen,

ich habe eine Frage zur Steuerung des Dynamic Power Limiters (DPL) über die Web-API. Ist es möglich, die Werte für den DPL-Modus, wie bei MQTT mit den Einstellungen Mode 1 / 2 / 3, direkt über die Web-API abzufragen und zu setzen? Oder ist das Ein- und Ausschalten des Dynamic Power Limiters tatsächlich nur über den Endpunkt /settings/powerlimiter möglich, sodass beim Aktivieren alle relevanten Werte gleichzeitig mitgesendet werden müssen?

Vielen Dank im Voraus für eure Hilfe!

**Comment by schlimmchen** [comment]
> sodass beim Aktivieren alle relevanten Werte gleichzeitig mitgesendet werden müssen?

So ist es. Dazu auch die [Important Note in der Doku](https://opendtu-onbattery.net/firmware/web_api/) nochmal verinnerlichen, bitte.

Geschickt ist es, wenn du dir das JSON mit den aktuellen Einstellungen holst, dann den Wert änderst, den du ändern willst, und dann das neue JSON verschickst (sorry, Endpunkte hab ich gerade nicht im Kopf). Weißt du, was ich meine?

**Reply by mastertape** [reply]
THX für die schnelle klarstellung 
Ja - die Notiz dazu haben wir verinnerlicht... und genau  so haben wir es ja testweise schon recht schnell erfolgreich hinbekommen den dpl zu deaktivieren / aktivieren...

Wir versuchen gerade auf tasmota basis mit berryscript ein billiges china display als fernsteuerung  ans laufen zu bekommen ohne nutzung eines mptt brokers ...

Unter mqtt gibt. es ja die 3 dpl modes - dazu gibt es aber in der webapi nix und somit auch nichts zum  ansprechen ...  ich hatte die hoffnung das es vielleicht doch eine  Möglichkeit gibt die nur nicht dokumentiert ist.. War quasi auf der Suche das Problem zu umgehen immer alle DPL Einstellungen mit senden zu müssen.. Wenn man da was ändert würde es bei nutzung  der Fernbedinung direkt überschrieben... (touchdisplay button und vorgefertigter Befehl wird gesendet) 

momentan ist alles sehr statisch ob es mit berry script möglich ist die einstellungen zu holen gut zwischen zu speichen usw. k.a. es ist derzeit ein erster versuch

**Reply by spcqike** [reply]
Kleine Anmerkung: wenn die Daten per Web-api gesendet werden, erfolgt jede Änderung persistent. Also mit schreib Zugriff auf den Flash. 

Per MQTT wäre es nur temporär und ohne Flash Zugriff.

**Reply by mastertape** [reply]
mqtt wäre um einiges einfacher umzusetzen - aber wenn vor ort kein mqtt serverbetrieben wird ... muss es so gehen...

**Reply by spcqike** [reply]
Und einfach einen machen? Auf nem Pi, notfalls auf nem weiteren esp? 

https://github.com/martin-ger/uMQTTBroker
